lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -290,6 +317,38 @@ def assert_valid_attributes!(attributes)
           raise ArgumentError, "Provided argument contains invalid keys, valid keys are #{PolicyElement.column_names}"
         end
       end
+
+      # Returns a hash containing the ancestors of the root nodes, with the interstitial
+      # ancestor relationships represented by the key/value pairs.
+      def get_ancestor_id_tree(root_nodes)
+        Assignment.select_ancestor_ids(root_nodes).each_with_object({}) do |row, memo|
+          id_array = row['ancestor_ids'].tr('{}','').split(',')
+          id_array.each { |ancestor_id| memo[ancestor_id] ||= [] }
```

Comment:
What's the impetus for this instantiation? It seems like we're going to overwrite it anyway. Is there a situation that would arise where a key in the `ancestor_id_tree` has no relationships?

Reply:
The situation where the ancestor has no ancestors of its own, i.e. it's a leaf node.